src/main/java/uk/gov/pay/ledger/eventpublisher/TopicNameArnMapper.java

```diff
@@ -2,20 +2,19 @@
 
 import com.google.inject.Inject;
 import uk.gov.pay.ledger.app.LedgerConfig;
+import uk.gov.pay.ledger.app.config.SnsConfig;
 
 import java.util.Map;
 
 class TopicNameArnMapper {
-    private final Map<TopicName, String> nameToArnMap;
+    private final SnsConfig snsConfig;
 
     @Inject
     TopicNameArnMapper(LedgerConfig ledgerConfig) {
-        this.nameToArnMap  = Map.of(
-                TopicName.CARD_PAYMENT_EVENTS,
-                ledgerConfig.getSnsConfig().getCardPaymentEventsTopicArn()
-        );
+        this.snsConfig  = ledgerConfig.getSnsConfig();
     }
     public String getArnForTopicName(TopicName topicName) {
-        return nameToArnMap.get(topicName);
+        return Map.of(TopicName.CARD_PAYMENT_EVENTS, snsConfig.getCardPaymentEventsTopicArn())
+                .get(topicName);
```

Comment:
It might be helpful if this threw a relevant error or null if no topic ARN was configured in the environment, which would then be handled further up. I don't think it's a deal breaker as this code can only be reached by setting a separate explicit env flag

Reply:
This also instantiates a map and looks up config on every publish (which will have a significant throughput) so could also be optimised by removing it from each call